src/app/debugger.js

```diff
@@ -16,19 +16,51 @@ function Debugger (id, editor, compiler, executionContextEvent, switchToFile, of
   this.editor = editor
   this.switchToFile = switchToFile
   this.compiler = compiler
+  this.markers = {}
+  this.breakPointManager = new remix.code.BreakpointManager(this.debugger, (sourceLocation) => {
+    return this.offsetToLineColumnConverter.offsetToLineColumn(sourceLocation, sourceLocation.file, this.editor, this.compiler.lastCompilationResult.data)
+  })
+
+  this.debugger.setBreakpointManager(this.breakPointManager)
+  this.breakPointManager.event.register('breakpointHit', (sourceLocation) => {})
+
+  function convertSourceLocation (self, fileName, row) {
+    var source = {}
+    for (let file in self.compiler.lastCompilationResult.data.sourceList) {
+      if (self.compiler.lastCompilationResult.data.sourceList[file] === fileName) {
+        source.file = file
+        break
+      }
+    }
+    source.row = row
+    return source
+  }
+
+  editor.event.register('breakpointCleared', (fileName, row) => {
+    if (self.compiler.lastCompilationResult.data) {
+      this.breakPointManager.remove(convertSourceLocation(this, fileName, row))
```

Comment:
The breakpoints should be file name based, the index is only valid for a single compilation run, but the breakpoints are not recreated.